WcaOnRails/app/views/competitions/index.html.erb

```diff
@@ -1,17 +1,144 @@
-    <% @competitions.each do |competition| %>
-      <tr>
-        <td><%= competition.id %></td>
-        <td><%= competition.name %></td>
-        <td><%= link_to "Edit", admin_edit_competition_path(competition) %></td>
-        <td><%= link_to "Results", "/results/competition.php?competitionId=#{competition.id}" %></td>
-      </tr>
+  <h2><%= yield(:title) %></h2>
+
+  <%= form_tag competitions_path, method: :get, class: "competition-select" do %>
+    <% if @competitions.length < 1 %>
+      <%= render 'index_form' %>
+      <div class="alert alert-danger">No competitions found!</div>
+    <% else %>
+      <% if params[:commit] == "List" %>
+        <table class="table competitions-table table-striped table-hover table-condensed floatThead">
+          <thead>
+            <tr>
+              <th colspan="7">
+                <%= render 'index_form' %>
+              </th>
+            </tr>
+            <tr>
+              <th>Year</th>
+              <th>Date</th>
+              <th>Name</th>
+              <th></th>
+              <th>Country, City</th>
+              <th>Venue</th>
+              <th class="big-column"></th>
+            </tr>
+          </thead>
+          <tbody>
+            <% @competitions.each_with_index do |competition, index| %>
+              <% if index == @closest_index %>
+                <tr id="today-line">
+                  <td colspan="6" class="text-center">Today</td>
+                  <td></td>
+                </tr>
+              <% end %>
+              <tr class="<%= competition.is_over? ? "past" : "future" %>">
+                <td><%= competition.year %></td>
+                <td><%= date_range(competition.start_date, competition.end_date, separator: '-', show_year: false) %></td>
+                <td>
+                  <% anchor_tag = @closest_index && (index == (@closest_index - 10)) %>
+                  <%= link_to competition.name, competition_path(competition), id: anchor_tag ? "today" : "" %>
+                </td>
+                <td>
+                  <%# results check... %>
+                </td>
+                <td><strong><%= competition.country.name %></strong>, <%= competition.cityName %></td>
+                <td><%=md(competition.venue, target_blank: true) %></td>
+                <td class="big-column"></td>
+              </tr>
+            <% end %>
+          </tbody>
+        </table>
+      <% else %>
+        <%= render 'index_form' %>
+        <div id="map-wrapper">
+          <div class="map"></div>
+        </div>
+      <% end %>
     <% end %>
-  </table>
+  <% end %>
 </div>
+
+<script>
+$(function() {
+  var $today = document.getElementById("today");
+  if ($today)
+    $(document).scrollTop( $("#today").offset().top );
+
+  function initMap() {
+    var $map = new google.maps.Map(document.getElementById('map-wrapper'), {
+      zoom: 2,
+      center: {lat: 0, lng: 0},
+      scrollwheel: true
+    });
+
+    wca.setCompetitions(<%= @competitions.map do |c|
```

Comment:
Indentation looks screwy here, but lets talk about this javascript on gchat, since I want to talk about https://github.com/cubing/worldcubeassociation.org/pull/319/files#r51315220